### ISSUE TITLE
Certain database rows cause prisma client to crash

### ISSUE THREAD

**Issue by comsma** [prisma/prisma#15445]
### Bug description

When running the Prisma.model.findMany command one of my rows in sql causes Prisma to crash and output this error. Im not entirely sure if there's a malformed value in my database. Im also not 100% sure what this error is trying to tell me or what column is causing this error.

```
thread 'tokio-runtime-worker' panicked at 'Given exponent overflowing the maximum accepted scale (255).: TryFromIntError(())', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tiberius-0.9.3/src/tds/numeric.rs:287:47
stack backtrace:
   0:        0x114188924 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h188b7ef1c7993e78
   1:        0x1141a8020 - core::fmt::write::he84a3004e7af3f34
   2:        0x114182bf4 - std::io::Write::write_fmt::h9370b50affaab0be
   3:        0x11418a070 - std::panicking::default_hook::{{closure}}::hc074f8023cce83ca
   4:        0x114189dd8 - std::panicking::default_hook::hef854b51b9b79ff2
   5:        0x11418a508 - std::panicking::rust_panic_with_hook::h1e59e224d558a492
   6:        0x11418a43c - std::panicking::begin_panic_handler::{{closure}}::he1a9d6ab32bfd8c6
   7:        0x114188e00 - std::sys_common::backtrace::__rust_end_short_backtrace::he9b94791b02f48cd
   8:        0x11418a194 - _rust_begin_unwind
   9:        0x11420f964 - core::panicking::panic_fmt::h9fec86f6a9c4146e
  10:        0x11420fa08 - core::result::unwrap_failed::h04f08301b97a771c
  11:        0x113dc9690 - tiberius::tds::numeric::bigdecimal_::<impl tiberius::to_sql::ToSql for core::option::Option<bigdecimal::BigDecimal>>::to_sql::he5f2c3bc35de62b0
  12:        0x113a3e4f4 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h1bb2b6a9432cac5a
  13:        0x113b4b484 - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::h2c448aba16cbe0dd
  14:        0x113a45154 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h2d24b4e1de521fc1
  15:        0x113a67c5c - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hd131c838039a4de9
  16:        0x113a426a0 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h29d081eee69d4353
  17:        0x1135e9cb0 - <tracing_futures::Instrumented<T> as core::future::future::Future>::poll::h6a0e88c60196da0f
  18:        0x1135c61b8 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h23c0ec236bd0df8d
  19:        0x1135bfabc - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h18a32ffe68cbd829
  20:        0x1135d38c8 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h9e8ceff94408aad4
  21:        0x1132b68f4 - query_core::interpreter::query_interpreters::read::read_related::{{closure}}::h14363e55dbc2eaee
  22:        0x1132a8d10 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h378feeca0e4e39a2
  23:        0x1132ab050 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h8100932631ec4597
  24:        0x1132a3ba4 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h07f4ac0fa08d642e
  25:        0x1132aa3a8 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7db6055ee109787b
  26:        0x1132ab080 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h8100932631ec4597
  27:        0x11326c824 - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::h355e260569f4cfa6
  28:        0x1132a3568 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h07796cc1a1e899eb
  29:        0x1132a92d0 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7106db4746c0e007
  30:        0x1130c38e0 - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::hf1dd5c4c78895573
  31:        0x113072cc0 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hb5050e51c5321afd
  32:        0x11306f674 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h997271f988289b2d
  33:        0x11307e024 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hd10e9bfc8e75506a
  34:        0x113064358 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h361ae0ceb45fab66
  35:        0x11306a390 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h73b5b997c96f0f75
  36:        0x11306c71c - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h7d8bfc95225ffe44
  37:        0x1130c35e0 - <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll::h9106f7b9f1dc9491
  38:        0x113078594 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hbdf923899c928647
  39:        0x113081c60 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hde71dadbf14e540e
  40:        0x1130670cc - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h5b7ac248ee7f63fd
  41:        0x1131306f0 - tokio::runtime::task::core::CoreStage<T>::poll::hffd92748427be685
  42:        0x11300929c - tokio::runtime::task::harness::poll_future::h0b21672fa4bef1c1
  43:        0x11300d85c - tokio::runtime::task::harness::Harness<T,S>::poll::h21d125b77f2c8822
  44:        0x113dec290 - std::thread::local::LocalKey<T>::with::h9edccece6c2f3963
  45:        0x113e027cc - tokio::runtime::thread_pool::worker::Context::run_task::h3f0afe5218ac3f54
  46:        0x113e01b18 - tokio::runtime::thread_pool::worker::Context::run::h0aab08c900c3bd18
  47:        0x113defbac - tokio::macros::scoped_tls::ScopedKey<T>::set::h28c8de06c0a15374
  48:        0x113e015e0 - tokio::runtime::thread_pool::worker::run::hf7b13b0164905c27
  49:        0x113df8c00 - <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll::hc3991b35707512ae
  50:        0x113df9f08 - tokio::runtime::task::harness::Harness<T,S>::poll::h0e33bd385a4898d3
  51:        0x113de4cf8 - tokio::runtime::blocking::pool::Inner::run::h55cea32f5020a611
  52:        0x113debf54 - std::sys_common::backtrace::__rust_begin_short_backtrace::h756ca1dda46c8a00
  53:        0x113decb68 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h96e9be7d1b666a9d
  54:        0x11418e828 - std::sys::unix::thread::Thread::new::thread_start::h7b2f9b83fb320a20
  55:        0x19ff8026c - __pthread_deallocate

/Users/collinsmallegan/Projects/Materials Resources/Prophet21Connector/node_modules/@prisma/client/runtime/index.js:29917
      throw new PrismaClientRustPanicError(message, this.client._clientVersion);
            ^
Error: 
Invalid `this.prisma.inventory_receipts_hdr.findUnique()` invocation in
/Users/collinsmallegan/Projects/Materials Resources/Prophet21Connector/src/orders/orders.service.ts:10:47

   7 constructor(private prisma: PrismaService) {}
   8 
   9 async readPickTicket(request: readPickTicketRequest) {
→ 10   return this.prisma.inventory_receipts_hdr.findUnique(
Given exponent overflowing the maximum accepted scale (255).: TryFromIntError(())

This is a non-recoverable error which probably happens when the Prisma Query Engine has a panic.

```

### Expected behavior

Prism a shouldn't crash when running a query.

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}


model invoice_hdr {
  invoice_no   String  @id(map: "pk_invoice_hdr", clustered: false) @db.VarChar(10)
  ship2_state  String
  total_amount Decimal @default(0, map: "df_invoice_hdr_total_amount") @db.Decimal(19, 4)
}

model oe_pick_ticket {
  pick_ticket_no Decimal @id(map: "pk_oe_pick_ticket") @db.Decimal(19, 0)
}

model oe_hdr {
  order_no              String  @id(map: "pk_oe_hdr") @db.VarChar(8)
  delivery_instructions String?  @db.VarChar(255)
  oe_hdr_uid            Int     @unique(map: "ak_oe_hdr_uid")
  ship2_add1            String? @db.VarChar(50)
  ship2_add2            String? @db.VarChar(50)
  ship2_country         String? @db.VarChar(50)
  ship2_city            String? @db.VarChar(50)
  ship2_state           String? @db.VarChar(50)
  ship2_zip             String? @db.VarChar(10)
}

model inventory_receipts_hdr {
  receipt_number Decimal @id(map: "pk_inventory_receipts_hdr") @db.Decimal(19, 0)
  inventory_receipts_line inventory_receipts_line[]
}

model inventory_receipts_line {
  receipt_number Decimal @db.Decimal(19, 0)
  line_number    Int

  inv_mast_uid    Int
  qty_received    Decimal @db.Decimal(19, 9)
  unit_of_measure String  @db.VarChar(8)

  //inv_mast                       inv_mast                         @relation(fields: [inv_mast_uid], references: [inv_mast_uid], onUpdate: NoAction, map: "FK50ofotx2q7fdwomayg9kvuss2")
  inventory_receipts_hdr inventory_receipts_hdr @relation(fields: [receipt_number], references: [receipt_number], onUpdate: NoAction)

  @@id([receipt_number, line_number], map: "pk_inventory_receipts_line", clustered: false)
}
```

```ts
return await this.prisma.inventory_receipts_hdr.findUnique({
      where: {
        receipt_number: receiptId,
      },
      include: {
        inventory_receipts_line: true,
      },
    });
  }
```

### Environment & setup

- OS: MacOS
- Database: Microsoft SQL Server
- Node.js version: v18.6.0

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 4.3.1
@prisma/client          : 4.3.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt c875e43600dfe042452e0b868f7a48b817b9640b (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.3.0-32.c875e43600dfe042452e0b868f7a48b817b9640b
Default Engines Hash    : c875e43600dfe042452e0b868f7a48b817b9640b
Studio                  : 0.473.0

```

**Comment by janpio** [comment]
I would guess it is getting some numeric value from the database that it can not put into the expected data structure.

Can you shared this query?
```
Invalid `this.prisma.inventory_receipts_hdr.findUnique()` invocation in
/Users/collinsmallegan/Projects/Materials Resources/Prophet21Connector/src/orders/orders.service.ts:10:47
```

From looking at the model:
```
model inventory_receipts_hdr {
  receipt_number Decimal @id(map: "pk_inventory_receipts_hdr") @db.Decimal(19, 0)
  inventory_receipts_line inventory_receipts_line[]
}
```

It could be that the `recpeipt_number` has some invalid values maybe that are not really `Decimal(19,0)`?

**Comment by comsma** [comment]
It only crashes when I use the include statement. Without using it and running a query for inventory_receipts_line all of my queries run 

Prisma inventory_receipts_line.findUnique returns the following without issue. My issue is trying to run 
```
return this.prisma.inventory_receipts_hdr.findUnique({
      include: {
        inventory_receipts_line: true,
      },
      where: {
        receipt_number: request.pickNo,
      },
    });
```
This query works without issue 
```
async readPickTicket(request: readPickTicketRequest) {
    return this.prisma.inventory_receipts_line.findMany({
      where: {
        receipt_number: request.pickNo,
      },
    });
```
```
[
  {
    receipt_number: 5000010,
    line_number: 1,
    inv_mast_uid: 19098,
    qty_received: 50,
    unit_of_measure: 'EA'
  }
]
[
  {
    receipt_number: 5000009,
    line_number: 1,
    inv_mast_uid: 19965,
    qty_received: 24,
    unit_of_measure: 'EA'
  }
]
[
  {
    receipt_number: 5000010,
    line_number: 1,
    inv_mast_uid: 19098,
    qty_received: 50,
    unit_of_measure: 'EA'
  }
]
[
  {
    receipt_number: 5000011,
    line_number: 1,
    inv_mast_uid: 24917,
    qty_received: 1000,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 2,
    inv_mast_uid: 24783,
    qty_received: 6000,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 3,
    inv_mast_uid: 24918,
    qty_received: 30000,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 4,
    inv_mast_uid: 38062,
    qty_received: 2000,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 5,
    inv_mast_uid: 24874,
    qty_received: 120,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 6,
    inv_mast_uid: 24921,
    qty_received: 100,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 7,
    inv_mast_uid: 24885,
    qty_received: 6,
    unit_of_measure: 'EA'
  },
  {
    receipt_number: 5000011,
    line_number: 8,
    inv_mast_uid: 24934,
    qty_received: 100,
    unit_of_measure: 'EA'
  }
]
```

**Comment by janpio** [comment]
> Is it possible that Prisma is trying to process data that isn't listed in the schema? As there is more columns in the table that are not utilized by Prisma?

Yes, that is why I asked for the full query - but I now saw that you had provided it, and it just got lost via the code formatting in GitHub. Back oot the query: An `include` indeed pulls in another model, in your case `inventory_receipts_line`:

```
model inventory_receipts_line {
  receipt_number Decimal @db.Decimal(19, 0)
  line_number    Int

  inv_mast_uid    Int
  qty_received    Decimal @db.Decimal(19, 9)
  unit_of_measure String  @db.VarChar(8)

  inventory_receipts_hdr inventory_receipts_hdr @relation(fields: [receipt_number], references: [receipt_number], onUpdate: NoAction)

  @@id([receipt_number, line_number], map: "pk_inventory_receipts_line", clustered: false)
}
```

I think the last Json object you provided above is that data already? Can you confirm that?

Do I understand that the only line item for 5000010 is this one:
```
[
  {
    receipt_number: 5000010,
    line_number: 1,
    inv_mast_uid: 19098,
    qty_received: 50,
    unit_of_measure: 'EA'
  }
]
```

And still requesting with `where = receipt_number: 5000010` it crashes?

**Comment by comsma** [comment]
I'm using Prisma onto of a erp system hence why I only implemented the columns I needed and not everything. Here is a JSON object for receipt_number 5000009 - 5000011. Both 5000009 and 5000010 give a single object response whereas 5000011 is a multi object response. Yet Prisma crashes with receipt_number=5000010.

```JSON
[
  {
    "receipt_number": 5000009,
    "line_number": 1,
    "qty_received": 24.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 24.000000000,
    "date_created": "2005-10-31 12:20:59.000",
    "date_last_modified": "2005-11-02 15:54:09.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 24.000000000,
    "po_line_number": 1,
    "unit_cost": 3.250000000,
    "extended_cost": 78.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 78.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 3.250000000,
    "extended_cost_display": 78.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 19965,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 81387,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000010,
    "line_number": 1,
    "qty_received": 50.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 50.000000000,
    "date_created": "2005-10-31 12:23:23.000",
    "date_last_modified": "2005-11-09 11:35:00.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 50.000000000,
    "po_line_number": 1,
    "unit_cost": 17.000000000,
    "extended_cost": 850.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 850.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 17.000000000,
    "extended_cost_display": 850.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 19098,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 77975,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 1,
    "qty_received": 1000.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 1000.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 1000.000000000,
    "po_line_number": 1,
    "unit_cost": 0.310000000,
    "extended_cost": 310.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 310.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.310000000,
    "extended_cost_display": 310.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24917,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 121760,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 2,
    "qty_received": 6000.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 6000.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 6000.000000000,
    "po_line_number": 2,
    "unit_cost": 0.292000000,
    "extended_cost": 1752.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 1752.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.292000000,
    "extended_cost_display": 1752.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24783,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 119041,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 3,
    "qty_received": 30000.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 30000.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 30000.000000000,
    "po_line_number": 3,
    "unit_cost": 0.290000000,
    "extended_cost": 8700.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 8700.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.290000000,
    "extended_cost_display": 8700.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24918,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 121782,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 4,
    "qty_received": 2000.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 2000.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 2000.000000000,
    "po_line_number": 4,
    "unit_cost": 0.250000000,
    "extended_cost": 500.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 500.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.250000000,
    "extended_cost_display": 500.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 38062,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 169877,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 5,
    "qty_received": 120.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 120.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 120.000000000,
    "po_line_number": 5,
    "unit_cost": 0.393000000,
    "extended_cost": 47.1600,
    "freight_amount": 0.0000,
    "amount_vouched": 47.1600,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.393000000,
    "extended_cost_display": 47.1600,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24874,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 120751,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 6,
    "qty_received": 100.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 100.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 100.000000000,
    "po_line_number": 6,
    "unit_cost": 0.520000000,
    "extended_cost": 52.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 52.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 0.520000000,
    "extended_cost_display": 52.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24921,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 121846,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 7,
    "qty_received": 6.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 6.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 6.000000000,
    "po_line_number": 8,
    "unit_cost": 11.950000000,
    "extended_cost": 71.7000,
    "freight_amount": 0.0000,
    "amount_vouched": 71.7000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 11.950000000,
    "extended_cost_display": 71.7000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24885,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 121387,
    "receipt_removed_by": null
  },
  {
    "receipt_number": 5000011,
    "line_number": 8,
    "qty_received": 100.000000000,
    "unit_of_measure": "EA",
    "unit_size": 1.0000,
    "unit_quantity": 100.000000000,
    "date_created": "2005-10-31 12:29:40.000",
    "date_last_modified": "2005-11-03 14:12:30.000",
    "last_maintained_by": "LST",
    "vouch_complete": "Y",
    "qty_vouched": 100.000000000,
    "po_line_number": 11,
    "unit_cost": 1.400000000,
    "extended_cost": 140.0000,
    "freight_amount": 0.0000,
    "amount_vouched": 140.0000,
    "freight_amount_vouched": 0.0000,
    "unit_cost_display": 1.400000000,
    "extended_cost_display": 140.0000,
    "freight_amount_display": 0.0000,
    "inv_mast_uid": 24934,
    "qty_disassembled": 0.000000000,
    "qty_lost": null,
    "msp_landed_cost": 0.000000000,
    "substitute_item": "N",
    "original_inv_mast_uid": null,
    "pricing_unit": "EA",
    "pricing_unit_size": 1.0000,
    "substitute_line_number": null,
    "date_receipt_removed": null,
    "retrieved_by_wms": null,
    "transfer_flag": null,
    "country_of_origin": null,
    "vessel_landed_cost_posted_amt": null,
    "wrong_part_no_flag": "N",
    "pallet_id": null,
    "created_by": "MRS\\Administrator",
    "label_qty": null,
    "inventory_receipts_line_uid": 122105,
    "receipt_removed_by": null
  }
]
```

This is the DDL in SqlServer
```SQL
-- auto-generated definition
create table inventory_receipts_line
(
    receipt_number                decimal(19)                                  not null
        constraint fk_inv_rcpt_line_inv_rcpt_hdr
            references inventory_receipts_hdr,
    line_number                   int                                          not null,
    qty_received                  decimal(19, 9)                               not null,
    unit_of_measure               varchar(8)                                   not null,
    unit_size                     decimal(19, 4)                               not null,
    unit_quantity                 decimal(19, 9)                               not null,
    date_created                  datetime                                     not null,
    date_last_modified            datetime                                     not null,
    last_maintained_by            varchar(30)                                  not null,
    vouch_complete                char                                         not null,
    qty_vouched                   decimal(19, 9)                               not null,
    po_line_number                int                                          not null,
    unit_cost                     decimal(19, 9),
    extended_cost                 decimal(19, 4),
    freight_amount                decimal(19, 4),
    amount_vouched                decimal(19, 4),
    freight_amount_vouched        decimal(19, 4),
    unit_cost_display             decimal(19, 9),
    extended_cost_display         decimal(19, 4),
    freight_amount_display        decimal(19, 4),
    inv_mast_uid                  int                                          not null,
    qty_disassembled              decimal(19, 9) default 0                     not null,
    qty_lost                      decimal(19, 9),
    msp_landed_cost               decimal(19, 9),
    substitute_item               char           default 'N'                   not null,
    original_inv_mast_uid         int,
    pricing_unit                  varchar(8)                                   not null,
    pricing_unit_size             decimal(19, 4)                               not null,
    substitute_line_number        int,
    date_receipt_removed          datetime,
    retrieved_by_wms              char
        constraint df_inventory_receipts_line_retrieved_by_wms default 'N',
    transfer_flag                 char,
    country_of_origin             varchar(255),
    vessel_landed_cost_posted_amt decimal(19, 4),
    wrong_part_no_flag            char
        constraint df_inventory_receipts_line_wrong_part_no_flag default 'N'   not null,
    pallet_id                     varchar(30),
    created_by                    varchar(255)
        constraint df_inventory_receipts_line_created_by default suser_sname() not null,
    label_qty                     int,
    inventory_receipts_line_uid   int identity,
    receipt_removed_by            varchar(255),
    constraint pk_inventory_receipts_line
        primary key nonclustered (receipt_number, line_number)
            with (fillfactor = 80)
)
go

create clustered index idx_inv_recpts_line_covered
    on inventory_receipts_line (inv_mast_uid, receipt_number, line_number, date_created)
    with (fillfactor = 80)
go

create index idx_inv_recpt_line_wms
    on inventory_receipts_line (retrieved_by_wms)
    with (fillfactor = 80)
go

create index idx_inventory_receipts_line_vendor_scorecard
    on inventory_receipts_line (date_created, receipt_number, po_line_number, qty_received)
    with (fillfactor = 80)
go

grant delete, insert, references, select, update on inventory_receipts_line to PxxiUser
go

grant delete, insert, references, select, update on inventory_receipts_line to p21_application_role
go


```

**Comment by janpio** [comment]
Befrore we spend more time on this: This could already be fixed in `dev` versions of 4.4.0, see https://github.com/prisma/prisma/issues/15079 and https://github.com/prisma/prisma/issues/14703. Can you try installing `prisma@dev` and `@prisma/client@dev` in your project and see if this fixes the query in question? Then this fix would be included in the release next Tuesday.

**Comment by comsma** [comment]
That seemed to fix it I also added a map key.

```
inventory_receipts_hdr inventory_receipts_hdr @relation(fields: [receipt_number], references: [receipt_number], onUpdate: NoAction, map: "fk_inv_rcpt_line_inv_rcpt_hdr")
```
